src/coordinator/executor/transform/types.go

```diff
@@ -46,3 +46,23 @@ type TimeSpec struct {
 	Now  time.Time
 	Step time.Duration
 }
+
+// Params are defined by transforms
+type Params interface {
+	parser.Params
+	Node(controller *Controller) OpNode
+}
+
+// SeriesNode is implemented by function nodes which can support series iteration
```

Comment:
Worth having a common Node interface that these both implement?
```golang
type Node interface{
	Meta(meta block.Metadata) block.Metadata
 	SeriesMeta(metas []block.SeriesMeta) []block.SeriesMeta
}
```

Reply:
Also, worth having an interface that matches functions that support both?

Reply:
common interface makes sense. I thought about the third interface matching both but didn't really find much uses for it. Might revisit later if we need it.